Src/DependencyCollector/Shared/Implementation/ActiveSubsciptionManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Linq;
+    using System.Threading;
+
+    /// <summary>
+    /// Keeps one active subscription to specific DiagnosticSource per process.
+    /// Helps manage subsciptions in scenarios where multiple apps hosted in the same process.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ActiveSubsciptionManager
```

Comment:
Does it have to be public (even with `EditorBrowsableState.Never`)?

Reply:
yes, as I'm going to reuse it in AspNetCore SDK for request deduplication